src/transmitter/index.ts

```diff
@@ -105,3 +107,38 @@ async function retryRequest(verb: NeedleHttpVerbs, url: string, payload: object)
 
   return {response, attempt};
 }
+
+/**
+ * Exported for testing
+ */
+export function getProxy(
+  config: Record<string, any>,
+  endpoint: string
+): string | undefined | never {
```

Comment:
what is the `never` part?

Reply:
Being explicit that this function may throw, maybe that's something that only I would take note of 😆 
In languages like Java you got explicit exceptions in the method signature e.g. `throws IllegalAccessException`; this `never` here is a hint that the function throws

Reply:
I'm not sure what's my opinion about that 😆 not sure that returning `never` will hint someone about an exception, for me it is a hint for an infinite loop

Reply:
https://www.typescriptlang.org/docs/handbook/basic-types.html#never
It could be either 😆 To avoid ambiguity I can just remove that, doesn't seem helpful if it only raises questions